test/unit/content-src/lib/selectLayoutRender.test.js

```diff
@@ -283,7 +282,7 @@ describe("selectLayoutRender", () => {
 
     assert.equal(layoutRender[0].components[0].type, "foo1");
     assert.equal(layoutRender[0].components[1].type, "foo2");
-    assert.equal(layoutRender[0].components[2], undefined);
+    assert.isTrue(layoutRender[0].components[2].data.recommendations[0].placeholder);
```

Comment:
Only one component (which causes early bail out)  has placeholder when feed isn't ready. Will be good to add assert showing layout render has only three components  and fourth component is undefined. 

assert.lengthOf(layoutRender[0].components, 3);
assert.isUndefined(layoutRender[0].components[3]);